### ISSUE TITLE
Implement the reformulated Raster Time Series specification

### ISSUE THREAD

**Issue by soaressgabriel** [LabSid-USP/RUBEM#181]
#### Raster Time Series Specifications

1. Time series of rasters explicitly defined with start and end dates for each raster (e.g. for $N$ months with $M$ rasters, where $M < N$):

```json
"raster_series": {
  "landuse": [
    {
      "file_path": "/home/Workspaces/maps/lulc/cob.map",
      "from": { "$ref": "#/simulation_period/start" },
      "to": "2034-12-31"
    },
    {
      "file_path": "/home/Workspaces/maps/lulc/cobpr.map",
      "from": "2035-01-01",
      "to": { "$ref": "#/simulation_period/finish" }
    }
  ],
},
```

2. Explicitly defined raster time series with 12 rasters to be repeated monthly in their respective month (e.g. for $N$ months with 12 rasters):

```json
"raster_series": {
  "ndvi": {
    "monthly": [
      {
        "month": 1,
        "file_path": "/home/Workspaces/maps/ndvi0000.001"
      },
      {
        "month": 2,
        "file_path": "/home/Workspaces/maps/ndvi0000.002"
      },
      {
        "month": 3,
        "file_path": "/home/Workspaces/maps/ndvi0000.003"
      },
      {
        "month": 4,
        "file_path": "/home/Workspaces/maps/ndvi0000.004"
      },
      {
        "month": 5,
        "file_path": "/home/Workspaces/maps/ndvi0000.005"
      },
      {
        "month": 6,
        "file_path": "/home/Workspaces/maps/ndvi0000.006"
      },
      {
        "month": 7,
        "file_path": "/home/Workspaces/maps/ndvi0000.007"
      },
      {
        "month": 8,
        "file_path": "/home/Workspaces/maps/ndvi0000.008"
      },
      {
        "month": 9,
        "file_path": "/home/Workspaces/maps/ndvi0000.009"
      },
      {
        "month": 10,
        "file_path": "/home/Workspaces/maps/ndvi0000.010"
      },
      {
        "month": 11,
        "file_path": "/home/Workspaces/maps/ndvi0000.011"
      },
      {
        "month": 12,
        "file_path": "/home/Workspaces/maps/ndvi0000.012"
      }
    ],
},
```

Optionally, it should be possible to define a threshold year to use a raster file for all following months:

```json
"raster_series": {
  "ndvi": {
    "monthly": [
      {
        "month": 1,
        "file_path": "/home/Workspaces/maps/ndvi0000.001"
      },
      //...
    ],
    "yearly_from": 2025,
    "yearly_file_path": "/home/Workspaces/maps/ndvipr.map"
  }
},
```

3. Time series of rasters defined by a directory with a common prefix for the files (e.g. for $N$ months with $M = N$ rasters):

```json
"raster_series": {
  "etp": {
    "dir_path": "/home/Workspaces/maps/etp/",
    "files_prefix": "etp"
  },
},
```

### Solution

- TBF

### Alternative solution(s)

- TBF

### Additional context

- See #126.

**Comment by soaressgabriel** [comment]
Reopening: still applies. The three specifications (list of date ranges, monthly with yearly repetition, directory series) will be implemented as part of the configuration v1.0 work tracked in #179, backed by a per-timestep path resolver in the model. Scheduled after the configuration layer migration.

**Comment by soaressgabriel** [comment]
Delivered in configuration format 1.0:
- #271 `ModelConfigurationFileV1` models the three raster series specifications of this issue (dated entries with `from`/`to`, the twelve `monthly` rasters with an optional yearly replacement, and directory series);
- #273 resolves every series to one raster per step through `DirectorySeriesResolver`, `DatedSeriesResolver` and `MonthlySeriesResolver` (bounds and overlap checks, `MissingStep` markers keeping the NDVI/land use fallback);
- #275 activates the format in the loader and on the command line (`rubem config schema`, `rubem config migrate`).

Validation of the resolved series (coverage of the simulated window, content rules, CRS) followed in #303, #305 and #310. Closing; the abandoned `feature/181-...` branch is deleted as part of #306.
